src/app.tsx

```diff
@@ -3,13 +3,23 @@ import Config from './components/config.jsx'
 import { ConfigContext } from './context/config-context.jsx'
 import HelperUi from './helper-ui.jsx'
 import { isConfigPage } from './lib/is-config-page.js'
-import { isPathOrSubdomainRequest } from './lib/path-or-subdomain.js'
+import { isPathOrSubdomainRequest, isSubdomainGatewayRequest } from './lib/path-or-subdomain.js'
 import RedirectPage from './redirectPage.jsx'
 
 function App (): JSX.Element {
   const { isConfigExpanded, setConfigExpanded } = useContext(ConfigContext)
   const isRequestToViewConfigPage = isConfigPage(window.location.hash)
-  const isSubdomainRender = isPathOrSubdomainRequest(window.location)
+  const isSubdomainRender = isSubdomainGatewayRequest(window.location)
+  const shouldRequestBeHandledByServiceWorker = isPathOrSubdomainRequest(window.location) && !isRequestToViewConfigPage
```

Comment:
this was being incorrectly set previously.

* `isPathOrSubdomainRequest` returns whether it is a path or subdomain request for ipfs/ipns content
* `isSubdomainRender` is now set to specifically represent if its a subdomain request
* `shouldRequestBeHandledByServiceWorker` is set to be `isPathOrSubdomainRequest` & NOT `isRequestToViewConfigPage`

TODO: we should abstract this logic into a function we can test.